plugins/org.jboss.tools.jst.js.bower/src/org/jboss/tools/jst/js/bower/internal/launch/shortcut/BowerUpdateCommandHandler.java

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  Contributors:
+ *       Red Hat, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.jboss.tools.jst.js.bower.internal.launch.shortcut;
+
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.debug.internal.ui.stringsubstitution.SelectedResourceManager;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.ui.IEditorPart;
+
+/**
+ * @author "Alexey Kazakov (alexeykazakov)"
+ */
+public class BowerUpdateCommandHandler extends AbstractHandler {
+
+	@Override
+	public Object execute(ExecutionEvent event) throws ExecutionException {
+		BowerUpdate update = new BowerUpdate();
+		IStructuredSelection ss = SelectedResourceManager.getDefault().getCurrentSelection();
```

Comment:
As I said above we should not probably require a selection at all. Just pick up the very first bower/npm project and run the corresponding launcher for it. But if the selection is available then use it (will help with multiple bower/npm projects).